### PR TITLE
Update issue templates to yml

### PR DESCRIPTION
**Pull request by andygup**
* Remove old markdown templates and replace with yml.
* Add issue verification step
* Add required fields
* Consolidate ESM and AMD md templates into a single issue yml template

cc @odoe @dasa

### CODE REVIEW

**Review by odoe — APPROVED** [review]
Nice update!

File: .github/ISSUE_TEMPLATE/issue-report.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Issue
+description: Report an issue with one of the samples in this repository. 
+labels: ["needs triage"]
+body:
+  - type: checkboxes
+    attributes:
+      label: IMPORTANT
+      description: Verify the following requirements have been met before opening an issue.
+      options:
+        - label: My question is related to the samples and content within this repository. For all other issues, please post your question in the [community forum](https://community.esri.com/t5/arcgis-api-for-javascript/ct-p/arcgis-api-for-javascript) or open a ticket with [Esri Technical Support](https://support.esri.com/).
```
Comment (bsvensson) [inline .github/ISSUE_TEMPLATE/issue-report.yml:10]: Should Support be the first option?

File: .github/ISSUE_TEMPLATE/issue-report.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          required: true
+        - label: I have [checked for existing issues](https://github.com/Esri/jsapi-resources/issues) to avoid duplicates. If someone has already opened an issue for what you are experiencing, please add a 👍 reaction to the existing issue instead of creating a new one.
```
Comment (bsvensson) [inline .github/ISSUE_TEMPLATE/issue-report.yml:12]: Only "reaction" or ... "and comment if necessary"?

File: .github/ISSUE_TEMPLATE/issue-report.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          required: true
+  - type: textarea
+    id: actual
+    attributes:
+      label: Actual Behavior
```
Comment (bsvensson) [inline .github/ISSUE_TEMPLATE/issue-report.yml:17]: Can/should we use "Sentence case" for all titles?

**Review by bsvensson — APPROVED** [review]
Looks great, just a few minor random comments :)

File: .github/ISSUE_TEMPLATE/issue-report.yml (hunk 1 above)
Comment (andygup) [reply .github/ISSUE_TEMPLATE/issue-report.yml:10]: Sounds good to me, done.

File: .github/ISSUE_TEMPLATE/issue-report.yml (hunk 3 above)
Comment (andygup) [reply .github/ISSUE_TEMPLATE/issue-report.yml:17]: Yep, done.

**Pull request merged**